community/blog/2020-12 Gain Insights in the Databricks Jobs with Arcus Background Jobs/README.md

```diff
@@ -0,0 +1,97 @@
+We'll use an Azure Function to create a timer that will periodically query this Databricks cluster.
+
+The job outcome itself will be written as an Application Insights Metric so we can use it later as a source of an alert.
+
+Here's an example of how this Azure Function can be set up:
+
+```csharp
+using Arcus.Security.Core;
+using Arcus.BackgroundJobs.Databricks;
+using Microsoft.Azure.Databricks.Client;
+
+namespace GainInsights
+{
+    public class DatabricksJobMetricsFunction
+    {
+        private readonly IConfiguration _configuration;
+        private readonly ISecretProvider _secretProvider;
+
+        public DatabricksJobMetricsFunction(IConfiguration configuration, ISecretProvider secretProvider)
+        {
+            _configuration = configuration;
+            _secretProvider = secretProvider;
+        }
+
+        [FunctionName("databricks-job-metrics")]
+        public async Task Run([TimerTrigger("0 */1 * * * *")]TimerInfo timer, ILogger logger)
+        {
+            var metricName = "Databricks Job Completed";
+            var baseUrl = _configuration.GetValue<string>("Arcus:Databricks:Url");
+            string secretToken = await _secretProvider.GetRawSecretAsync("Arcus.Databricks.SecretToken");
+
+            var startOfWindow = timer.ScheduleStatus.Last;
+            var endOfWindow = timer.ScheduleStatus.Next;
+
+            using var client = DatabricksClient.CreateClient(baseUrl, secretToken);
+            using (var provider = new DatabricksInfoProvider(client, logger))
+            {
+                await provider.MeasureJobOutcomesAsync(metricName, startOfWindow, endOfWindow);
+            }
+        }
+    }
+}
+```
+
+The `DatabricksInfoProvider` is an Arcus type that allows you to either measure the job outcomes directly, or query them so you can log them yourself. For more information on this type, see [the official docs](https://background-jobs.arcus-azure.net/features/databricks/gain-insights).
+
+Note that we specify a time window in which we want the completed jobs to exist. This is now set on the start and end of each run of the Azure Function time trigger.
+
+## Application Insights metrics and alerts
+
+All the previous setup  will result in log metrics like the following:
+```
+Metric Databricks Job Completed: 1 at 12/10/2020 08:50:01 +00:00 (Context: [Run Id, 77], [Job Id, 9], [Job Name, My Databricks Job], [Outcome, success])
+```
+
+If you want them to be in Application Insights, please use [our own Application Insights Serilog sink](https://observability.arcus-azure.net/features/sinks/azure-application-insights) so the written log message is discovered as metrics and not traces.
+
+The metrics will show up as custom metrics:
+
+![Custom metrics](./images/custom-metrics.png)
```

Comment:
You could also use the metric explorer to show the graph; if you want

Reply:
Yeah, but the metric values are always `1` in this case; not sure if that would add much of added-value. Showing the actual logs with the job outcome may be more appropriate here.

Reply:
Yes because you could sum() the values, but it's ok